lib/interactive_viewer.ml

```diff
@@ -60,9 +60,38 @@ let navigate z_patches (dir : direction) : unit =
   | Prev -> Lwd.set z_patches (Zipper.prev z)
   | Next -> Lwd.set z_patches (Zipper.next z)
 
-(* let pure_str s = Lwd.pure (W.string s) *)
 let quit = Lwd.var false
 
+let calculate_add_remove_operation z_patches : ui Lwd.t =
+  let$ z = Lwd.get z_patches in
+  let p = Zipper.get_focus z in
+  let additions, removals =
+    List.fold_left
+      (fun (add_acc, remove_acc) hunk ->
+        let calculate_add_remove_hunk lines =
```

Comment:
Including the function definition inside the `fold_left` is less readable and "redefines it" each time, in theory.